proto/public/api.proto

```diff
@@ -97,6 +97,10 @@ message ApiError {
   string error = 1;
 }
 
+message ApiErrors {
+  repeated ApiError errors = 1;
```

Comment:
Would you mind documenting exactly what `ApiError` is? It's not clear to me when it's appropriate to use this message type.

Reply:
Will do. TBH right now it's kind of a catch all; Really I just wanted the error map to be a `map<string, repeated string>`, so I created a `message Errors { repeated string errors = 1 }` and figured it might be confusing to have two types of errors in this manner. So I decided to reuse `ApiError`... but they're not actually for the same thing. This seems to indicate that maybe I should rename the array of string to be `Errors` or something other than `ApiErrors`....

Reply:
We should opt to use message types with the narrowest scope possible, rather than reuse a message type with ill-defined semantics. This is important for API extensibility -- if we overload message types, it makes them harder to change/extend later.

In this case, I'd probably prefer

```protobuf
message PodErrors {
  repeated PodError errors =1

  message PodError {
    oneof error {
      Unknown unknown = 1;
    }
    message Unknown {
      string message = 1;
    }
  }
}
```

That said, what I'd _really_ prefer from the API is an enumerated set of of pod errors that we handle.  If we can't enumerate them yet, I'd prefer to do something like the above so that we can easily get more specific later on.